src/Test/WebTestCase.php

```diff
@@ -218,7 +218,12 @@ protected function getContainer(): ContainerInterface
             $kernel = $this->createKernel($options);
             $kernel->boot();
 
-            $this->containers[$cacheKey] = $kernel->getContainer();
+            $container = $kernel->getContainer();
+            if ($container->has('test.service_container')) {
+                $this->containers[$cacheKey] = $container->get('test.service_container');
```

Comment:
Can be one-lined with a ternary:

```php
$this->containers[$cacheKey] = $kernel->getContainer()->has('test.service_container')
    ? $kernel->getContainer()->get('test.service_container')
    : $kernel->getContainer();
```

Reply:
Or

```
$container = $kernel->getContainer();
$this->containers[$cacheKey] = $container->has('test.service_container')
    ? $container->get('test.service_container')
    : $container;
```

To avoid repeating `$kernel->getContainer()`.